caffeine/src/main/java/com/percolate/caffeine/DialogUtils.java

```diff
@@ -31,13 +35,17 @@ public static AlertDialog quickDialog(final Activity context, final String messa
         final SpannableString s = new SpannableString(message); //Make links clickable
         Linkify.addLinks(s, Linkify.ALL);
 
-        Builder builder = new AlertDialog.Builder(context);
-        builder.setMessage(s);
-        builder.setPositiveButton(android.R.string.ok, closeDialogListener());
-        AlertDialog dialog = builder.create();
-        dialog.show();
-
-        ((TextView) dialog.findViewById(android.R.id.message)).setMovementMethod(LinkMovementMethod.getInstance()); //Make links clickable
+        if(dialog != null && dialog.isShowing()){
```

Comment:
Dialog for A will not dismiss / leak.  It would just be under Activity B.  If Activity A somehow goes away it would leak.